drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/utils/ScenarioSimulationBuilders.java

```diff
@@ -87,13 +87,17 @@ public ScenarioGridColumnBuilder setPlaceHolder(String placeHolder) {
         public ScenarioGridColumn build() {
             List<GridColumn.HeaderMetaData> headerMetaDataList = headerBuilder.build();
             ScenarioGridColumnRenderer actualScenarioGridColumnRenderer = scenarioGridColumnRenderer != null ? scenarioGridColumnRenderer : new ScenarioGridColumnRenderer();
-            return new ScenarioGridColumn(headerMetaDataList,
-                                                                 actualScenarioGridColumnRenderer,
-                                                                 width,
-                                                                 isMovable,
-                                                                 factoryCell,
-                                                                 placeHolder
+            ScenarioGridColumn scenarioGridColumn = new ScenarioGridColumn(headerMetaDataList,
+                                                                           actualScenarioGridColumnRenderer,
+                                                                           width,
+                                                                           isMovable,
+                                                                           factoryCell,
+                                                                           placeHolder
             );
+            // by default scenario columns should be auto to have auto resize
+            scenarioGridColumn.setColumnWidthMode(GridColumn.ColumnWidthMode.auto);
+            scenarioGridColumn.setMinimumWidth(width);
```

Comment:
If these are the defaults for ScenarioGridColumn, should we set them inside the ScenarioGridColumn constructor?

Reply:
Yes you are right, moved to the constructor